File: src/main/java/com/now/nowbot/qq/enums/Role.java
package com.now.nowbot.qq.enums;

public enum Role {
    OWNER, ADMIN, MEMBER;
    public static Role fromString(String s) {
        Role r;
        switch (s) {
            case "owner": r = OWNER;break;
            case "admin": r = ADMIN;break;
            case "member": r = MEMBER;break;
            default:r = MEMBER;
        }
        return r;
    }
}

File: src/main/java/com/now/nowbot/qq/enums/Role.kt
package com.now.nowbot.qq.enums

enum class Role {
    OWNER, ADMIN, MEMBER;

    companion object {
        fun getRole(s: String?): Role {
            return when (s?.trim()?.lowercase()) {
                "owner" -> OWNER
                "admin" -> ADMIN
                null -> MEMBER
                else -> MEMBER
            }
        }
    }
}